### PR TITLE
Fix handling plurals for languages with locale e.g. pl_PL, en_US

### PR DESCRIPTION
**Pull request by IShotTheSheriff**
Targets issue #288. 

With current plural handling we've been able to recognise ISO-692-2 language codes and set appropriate  plural forms based on it. However the matcher was failing silently if language code included locale data e.g. pl_PL, fr_FR, en_US etc. With this fix we're adding one extra step to make sure that those are stripped if necessary. 

From now on we're supporting any variation of codes, examples: pl, pl_PL, en_US, en, fr_FR, zh, sah etc. One single exception is Brazilian Portuguese as it differ from other Portuguese locale plural form and will be matched as pt_BR (without striping _BR). So if plural form is needed from 'pt' language code it has to be used either as strict 'pt' or with different locale like 'pt_PT', which is correct usage anyway.

### CODE REVIEW

**Comment by IShotTheSheriff** [comment]
I leave it as an pull request as it indeed adds one extra step in getting plural string. Most of my projects will definitely benefit and extra string operations shouldn't harm the performance as much I think. I leave it to your consideration though if it's worth it when you look on it from wider perspective.

File: src/plural.js [hunk 1]
```diff
@@ -110,5 +110,18 @@ angular.module("gettext").factory("gettextPlurals", function () {
...
+     * Method extracts iso639-2 language code from code with locale e.g. pl_PL, en_US, etc.
+     * If it's provided with standalone iso639-2 language code it simply returns it.
+     * @param {String} langCode
+     * @returns {String} iso639-2 language Code
+     */
+    function getLanguageCode(langCode) {
+        if( langCode === 'pt_BR' ) {
+            return 'pt_BR';
+        }
+        return langCode.split('_').shift();
```
Comment (rubenv) [inline src/plural.js]: You can reduce the cost of this method by adding a small cache within `gettextPlurals`:

```js
var languageCodes = {
    'pt_BR': 'pt_BR',
};
```

Then inside `getLanguageCode`:

```js
if (!languageCodes[langCode]) {
     languageCodes[langCode] = langCode.split('_').shift();
}
return languageCodes[langCode];
```

That should bring the extra cost of this down to almost zero.

**Comment by rubenv** [comment]
Looking good! I can't vouch for the correctness of your Polish examples, so I'm assuming they are correct. Great to have some testing on alternate plural-rules languages.

See my comment above about the caching, with that in place I see no reason why we shouldn't merge this, it'll make a lot of users very happy.

File: src/plural.js (hunk 1 above)
Comment (IShotTheSheriff) [inline src/plural.js]: Sure, thanks!

**Comment by IShotTheSheriff** [comment]
Thanks for the caching tip, already added it! And yea Polish examples are simply enough to be correct :).

**Comment by rubenv** [comment]
Merged, thanks a lot for looking into this, I've wanted this for quite some time but never got round to it.

I'm guessing there's gonna be quite a few issues we can go back to and close now.

Released as `v2.3.1`.

**Comment by IShotTheSheriff** [comment]
Sure no problem.

**Pull request merged**